query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1527,5 +1527,11 @@ void ExecutionGenerator::convertTableGenerator(
   temporary_relation_info_vec_.emplace_back(tablegen_index, output_relation);
 }
 
+S::InsertDestination* ExecutionGenerator::addNewInsertDestinationToQueryContext() {
+  std::unique_ptr<S::InsertDestination> insert_destination_proto(query_context_proto_->add_insert_destinations());
+  insert_destination_proto->set_query_id(query_context_proto_->query_id());
```

Comment:
InsertDestination needs the query ID because the ID needs to be put in the pipelining message sent by the InsertDestination. Now InsertDestination can be constructed from its proto representation. Therefore it is good to have it in the InsertDestination proto.